Sources/StreamChatUI/ChatMessageList/ChatMessage/RedPacketSentBubble.swift

```diff
@@ -216,87 +216,48 @@ class RedPacketSentBubble: UITableViewCell {
-            let one = fetchRawData(raw: maxOne) as? String ?? "0"
-            lblTotal.text = "Total: \(one) ONE"
+        descriptionLabel.text = extraData.redPacketTitle
+        if let maxOne = extraData.redPacketMaxOne {
+            lblTotal.text = "Total: \(maxOne) ONE"
         } else {
             lblTotal.text = "Total: 0 ONE"
         }
-        if let minOne = redPacket["minOne"] {
-            let one = fetchRawData(raw: minOne) as? String ?? "0"
-            lblMax.text = "Max: \(one) ONE"
+        if let minOne = extraData.redPacketMinOne {
+            lblMax.text = "Max: \(minOne) ONE"
         } else {
             lblMax.text = "Max: 0 ONE"
         }
-        if let participants = redPacket["participantsCount"] {
-            let participantCount = fetchRawData(raw: participants) as? String ?? "0"
-            let intParticipants = Int(participantCount) ?? 0
-            if intParticipants <= 1 {
-                lblDetails.text = "First user receives 100% of the packet!"
-            } else {
-                lblDetails.text = "Split randomly between: \(intParticipants) users"
-            }
-        }
-//        if let endTime = redPacket["endTime"] {
-//            let strEndTime = fetchRawData(raw: endTime) as? String ?? ""
-//            let dateFormatter = DateFormatter()
-//            dateFormatter.locale = Locale(identifier: "en_US_POSIX") // set locale to reliable US_POSIX
-//            dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
-//            if let date = dateFormatter.date(from: strEndTime) {
-//
-//            }
-//        }
-        lblExpire.text = "Expires in 30 minutes!"
-    }
-
-    private func getRedPacketExtraData() -> [String: RawJSON]? {
-        if let extraData = content?.extraData["redPacketPickup"] {
-            switch extraData {
-            case .dictionary(let dictionary):
-                return dictionary
-            default:
-                return nil
-            }
+        let participants = Int(extraData.redPacketParticipantsCount ?? "0") ?? 0
+        if participants <= 1 {
+            lblDetails.text = "First user receives 100% of the packet!"
         } else {
-            return nil
+            lblDetails.text = "Split randomly between: \(participants) users"
         }
+        lblExpire.text = "Expires in 30 minutes!"
     }
 
-    private func getEndTime(raw: [String: RawJSON]?) -> Date? {
-        guard let rawData = raw else { return nil }
-        if let endTime = rawData ["endTime"] {
-            let strEndTime = fetchRawData(raw: endTime) as? String ?? ""
-            let dateFormatter = ISO8601DateFormatter()
-            dateFormatter.formatOptions = [.withInternetDateTime]
-            if let date = dateFormatter.date(from: "\(strEndTime)") {
-                return date
-            } else {
-                return nil
-            }
+    private func getEndTime() -> Date? {
+        let strEndTime = content?.extraData.redPacketEndTime ?? ""
```

Comment:
@ajay-simform  can we add this into `DateFormatter+Extensions `file?

Reply:
@parth-simformsolutions  I will do that later as I am planning to refactor redPacket stuff.